server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -0,0 +1,635 @@
+    private static final String SENDER_1_ADDRESS = "sender1@james.org";
+    private static final String SENDER_1_USERNAME = "sender1";
+
+    private static final String SENDER_2_FULL_ADDRESS = "sender2 <sender2@james.org>";
+    private static final String SENDER_2_ADDRESS = "sender2@james.org";
+    private static final String SENDER_2_USERNAME = "sender2";
+
+    private static final String SENDER_1_AND_UNFOLDED_USER_FULL_ADDRESS = "sender1 <sender1@james.org>, \r\nunfolded\r\n_user\r\n <unfolded_user@james.org>";
+
+    private static final String RECIPIENT_TO_1_FULL_ADDRESS = "recipient_to_1 <recipient_to_1@james.org>";
+    private static final String RECIPIENT_TO_1_ADDRESS = "recipient_to_1@james.org";
+    private static final String RECIPIENT_TO_1_USERNAME = "recipient_to_1";
+
+    private static final String RECIPIENT_TO_2_FULL_ADDRESS = "recipient_to_2 <recipient_to_2@james.org>";
+    private static final String RECIPIENT_TO_2_ADDRESS = "recipient_to_2@james.org";
+    private static final String RECIPIENT_TO_2_USERNAME = "recipient_to_2";
+
+    private static final String RECIPIENT_CC_1_FULL_ADDRESS = "recipient_cc_1 <recipient_cc_1@james.org>";
+    private static final String RECIPIENT_CC_1_ADDRESS = "recipient_cc_1@james.org";
+    private static final String RECIPIENT_CC_1_USERNAME = "recipient_cc_1";
+
+    private static final String RECIPIENT_CC_2_FULL_ADDRESS = "recipient_cc_2 <recipient_cc_2@james.org>";
+
+    private static final String SCRAMBLED_SUBJECT = "this is the subject =?UTF-8?B?RnLDqWTDqXJpYyBNQVJUSU4=?= of the mail";
+    private static final String UNSCRAMBLED_SUBJECT = "this is the subject Frédéric MARTIN of the mail";
+    private static final String SHOULD_NOT_MATCH = "should not match";
+
+    private static final String RECIPIENT_TO_3_FULL_ADDRESS = "recipient_to_3 <recipient_to_3@james.org>";
+    private static final String RECIPIENT_TO_3_USERNAME = "recipient_to_3";
+    private static final String RECIPIENT_TO_4_FULL_ADDRESS = "recipient_to_4 <recipient_to_4@james.org>";
+
+    private static final String RECIPIENT_1 = "recipient1@james.org";
+    static final String RECIPIENT_1_USERNAME = "recipient1";
+    static final String RECIPIENT_1_MAILBOX_1 = "recipient1_maibox1";
+
+    private static final String FRED_MARTIN_FULLNAME = "Frédéric MARTIN";
+    private static final String FRED_MARTIN_FULL_SCRAMBLED_ADDRESS = "=?UTF-8?B?RnLDqWTDqXJpYyBNQVJUSU4=?= <fred.martin@linagora.com>";
+    private static final String FRED_MARTIN_FULL_ADDRESS = "Frédéric MARTIN <fred.martin@linagora.com>";
+
+    private static final String UNFOLDED_USERNAME = "unfolded_user";
+
+    private static final String SUFFIX = "suffix";
+
+    static Stream<Arguments> mailDirectiveShouldSetWhenContainsRuleValueParamsProvider() throws Exception {
+        return Stream.of(
+            Arguments.of("normal content", Rule.Condition.Field.FROM, mimeMessageBuilder().addFrom(SENDER_1_FULL_ADDRESS), SENDER_1_USERNAME),
```

Comment:
Idents arguments to make it readable?

Reply:
I would propose a builder to avoid passing a raw, not really readable, `Argument.of(...)`

Something like:

```
FilteringTestArument.builder()
    .comment("normal content")
    .field(FROM)
    .mimeMessage(...)
    .valueToMatch(SENDER_1_FULL_ADDRESS)
    .build();
```